src/core/border_router/routing_manager.cpp

```diff
@@ -1191,20 +1258,22 @@ bool RoutingManager::NetworkDataContainsOmrPrefix(const Ip6::Prefix &aPrefix) co
 
 // Update the `mRouterAdvMessage` with given Router Advertisement message.
 // Returns a boolean which indicates whether there are changes of `mRouterAdvMessage`.
-bool RoutingManager::UpdateRouterAdvMessage(const RouterAdv::RouterAdvMessage &aRouterAdvMessage)
+bool RoutingManager::UpdateRouterAdvMessage(const RouterAdv::RouterAdvMessage *aRouterAdvMessage)
 {
     RouterAdv::RouterAdvMessage oldRouterAdvMessage;
 
     oldRouterAdvMessage = mRouterAdvMessage;
-    if (aRouterAdvMessage.GetRouterLifetime() == 0)
+
+    mTimeRouterAdvMessageLastUpdate = TimerMilli::GetNow();
+    if (aRouterAdvMessage == nullptr || aRouterAdvMessage->GetRouterLifetime() == 0)
     {
         mRouterAdvMessage.SetToDefault();
     }
     else
     {
-        mRouterAdvMessage = aRouterAdvMessage;
-        // TODO: add a timer for invalidating the learned RA parameters
-        // for cases that the other RA daemon crashed or is force killed.
+        mRouterAdvMessage = *aRouterAdvMessage;
+        mDiscoveredPrefixStaleTimer.FireAtIfEarlier(mTimeRouterAdvMessageLastUpdate +
+                                                    Time::SecToMsec(kRtrAdvStaleTime));
```

Comment:
@jwhui We also update the STALE timer for the learned RA parameters so that we can always forget it even when the Wi-Fi daemon crashes without sending RA messages with zero router lifetime.